### PR TITLE
Course project listing and project view

### PR DESCRIPTION
**Pull request by nbuonin**
This PR implements a table on the course page to list projects. It also implements Bootstrap forms for the create and edit views for course projects. Lastly, it roughs in an implementation of the project page, including the map.

### CODE REVIEW

File: locustempus/main/tests/test_views.py [hunk 1]
```diff
@@ -1086,7 +1080,7 @@ def test_list_project_faculty(self):
             )
         )
         response = self.client.get(
-            "/course/{}/project/".format(self.sandbox_course.pk))
+            "/course/{}/".format(self.sandbox_course.pk))
```
Comment (nikolas) [inline locustempus/main/tests/test_views.py:1083]: It's better practice to use `reverse` to construct URLs in tests, like this: https://github.com/ccnmtl/mediathread/blob/master/mediathread/main/tests/test_views.py#L1619

Then you don't have to update everything here if the URLs change.

File: locustempus/main/views.py [hunk 2]
```diff
@@ -87,6 +87,13 @@ class CourseDetailView(LoggedInCourseMixin, DetailView):
     model = Course
     template_name = 'main/course_detail.html'
 
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        course: Course = kwargs.get('object')
```
Comment (nikolas) [inline locustempus/main/views.py:92]: Is this supposed to be `course = kwargs.get('object')`? I'm not sure if this is some new python syntax I don't know about...

Comment (nbuonin) [reply locustempus/main/views.py:92]: It's a type annotation. I'm trying out MyPy for this project. I forget the version, but as of Python 3.something it's valid Python. The interpreter ignores it at runtime.

Comment (nikolas) [reply locustempus/main/views.py:92]: ah okay, cool!

File: locustempus/main/tests/test_views.py (hunk 1 above)
Comment (nbuonin) [reply locustempus/main/tests/test_views.py]: Great point, I've replaced all the hard coded URLs.

**Pull request merged**